src/ontodia/data/sparql/sparqlDataProvider.ts

```diff
@@ -265,15 +339,46 @@ export class SparqlDataProvider implements DataProvider {
                 LIMIT ${params.limit} OFFSET ${params.offset}
             }
             ${resolveTemplate(this.settings.filterElementInfoPattern, {dataLabelProperty})}
+            ${blankNodeQuery}
         } ${textSearchPart ? 'ORDER BY DESC(?score)' : ''}
         `;
 
-        return this.executeSparqlQuery<ElementBinding>(query).then(getFilteredData);
+        return this.executeSparqlQuery<ElementBinding | BlankBinding>(query)
+            .then(result => {
+                if (this.options.acceptBlankNodes) {
+                    return BlankNodes.updateFilterResults(
+                        result,
+                        (callBackQuery: string) => {
+                            return this.executeSparqlQuery<BlankBinding>(callBackQuery);
+                        },
+                    );
+                } else {
+                    result.results.bindings = result.results.bindings
+                    return result;
+                }
+            }).then(getFilteredData);
     };
 
     executeSparqlQuery<Binding>(query: string) {
         const method = this.options.queryMethod ? this.options.queryMethod : SparqlQueryMethod.GET;
-        return executeSparqlQuery<Binding>(this.options.endpointUrl, query, method, this.options.queryFunction);
+        let promise = executeSparqlQuery<Binding>(this.options.endpointUrl, query, method, this.options.queryFunction);
+        return promise;
+    }
+
+    concatWithBlankNodeResponse<Binding>(
+        response: SparqlResponse<Binding>,
+        blankNodeResponse: SparqlResponse<Binding>,
+    ): SparqlResponse<Binding> {
+        if (!this.options.acceptBlankNodes) {
+            return response;
+        }
+        const resultsMixture: SparqlResponse<Binding> = {
+            head: { vars: response.head.vars },
+            results: { bindings: [] },
```

Comment:
You should create `resultsMixture` with already initialized `bindings` instead of doing another assignment later:
```
const bindings = blankNodeResponse.results.bindings.concat(response.results.bindings);
const resultsMixture: SparqlResponse<Binding> = {
    head: {vars: response.head.vars},
    results: {bindings},
};
```